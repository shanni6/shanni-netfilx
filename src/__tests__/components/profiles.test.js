import React from "react";
import { render } from "@testing-library/react";
import { Profiles } from "../../components";

describe("<Profiles />,", () => {
    it("renders the <Profiles /> with populated data", () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who is watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.Picture
                            src="/images/karl.png"
                            data-testid="profile-picture"
                        />
                        <Profiles.Name>Shanni6</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        );

        expect(getByText("Who is watching?"));
        expect(getByTestId("profile-picture")).toBeTruthy();
        expect(getByText("Shanni6"));
        expect(container.firstChild).toMatchSnapshot();
    });
    it("renders the <Profiles /> with populated data but misc profile images", () => {
        const { container, getByText, getByTestId } = render(
            <Profiles>
                <Profiles.Title>Who is watching?</Profiles.Title>
                <Profiles.List>
                    <Profiles.User onClick={() => {}}>
                        <Profiles.Picture data-testid="profile-picture-misc" />
                        <Profiles.Name>Shanni6</Profiles.Name>
                    </Profiles.User>
                </Profiles.List>
            </Profiles>
        );
        expect(getByText("Who is watching?"));
        expect(getByTestId("profile-picture-misc")).toBeTruthy();
        expect(getByText("Shanni6"));
        expect(container.firstChild).toMatchSnapshot();
    });
});
