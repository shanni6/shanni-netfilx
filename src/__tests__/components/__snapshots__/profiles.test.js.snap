// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles />, renders the <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-sc-bxtkkv-0 brFiEp"
>
  <h1
    class="profiles__Title-sc-bxtkkv-1 fjLfcR"
  >
    Who is watching?
  </h1>
  <ul
    class="profiles__List-sc-bxtkkv-2 etMhNa"
  >
    <li
      class="profiles__Item-sc-bxtkkv-5 gbPGck"
    >
      <img
        class="profiles__Picture-sc-bxtkkv-4 jJKsjl"
        data-testid="profile-picture"
        src="/images/users//images/karl.png.png"
      />
      <p
        class="profiles__Name-sc-bxtkkv-3 ecPTxa"
      >
        Shanni6
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles />, renders the <Profiles /> with populated data but misc profile images 1`] = `
<div
  class="profiles__Container-sc-bxtkkv-0 brFiEp"
>
  <h1
    class="profiles__Title-sc-bxtkkv-1 fjLfcR"
  >
    Who is watching?
  </h1>
  <ul
    class="profiles__List-sc-bxtkkv-2 etMhNa"
  >
    <li
      class="profiles__Item-sc-bxtkkv-5 gbPGck"
    >
      <img
        class="profiles__Picture-sc-bxtkkv-4 jJKsjl"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profiles__Name-sc-bxtkkv-3 ecPTxa"
      >
        Shanni6
      </p>
    </li>
  </ul>
</div>
`;
